core/commons/che-core-commons-logback/src/main/java/org/eclipse/che/commons/logback/filter/IdentityIdLoggerFilter.java

```diff
@@ -42,11 +42,12 @@ public final void doFilter(
       ServletRequest request, ServletResponse response, FilterChain filterChain)
       throws IOException, ServletException {
     final HttpServletRequest httpRequest = (HttpServletRequest) request;
-    final HttpSession session = httpRequest.getSession();
-    Subject subject = (Subject) session.getAttribute("che_subject");
-
-    if (subject != null && subject.getUserId() != null) {
-      MDC.put(IDENTITY_ID_MDC_KEY, subject.getUserId());
+    final HttpSession session = httpRequest.getSession(false);
+    if (session != null) {
+      Subject subject = (Subject) session.getAttribute("che_subject");
```

Comment:
can we take Subject from EnvironmentContext?

Reply:
Yes but only if we sure that thi filter is running any of auth ones,

Reply:
One way or another it should. The current implementation is also require that ```session.getAttribute("che_subject")```